corehq/apps/locations/queryutil.py

```diff
@@ -57,7 +57,8 @@ def __iter__(self):
                     items2 = list(self._cte_set)
                 ids1 = [_identify(it) for it in items1]
                 ids2 = [_identify(it) for it in items2]
-                if (finished and ids1 != ids2) or not ids1 == ids2[:len(ids1)]:
+                if (finished and Counter(ids1) != Counter(ids2)) or (
```

Comment:
`Counter` is a cool idea - looks like that's what unittest's `assertItemsEqual` uses too.

Might be worth a comment in-line here explaining what's going on.  Looks like this code makes a couple assumptions:
* Sometimes the return queryset has duplicate elements (is this true?)
* We don't need to check order.

What kind of situations would cause `finished` to equal `False` here?  In that event, why check that `ids2` has at least all the elements of `ids1`, should it be `set(ids1).symmetric_difference(set(ids2))`? (which I think can also be written `set(ids1) ^ set(ids2)`)
Looks like that'd happen when pulling from MPTT fails, is there another scenario where that might occur more frequently?

Reply:
> Might be worth a comment

d8398820d2a7b539ba912ebac48a33128fd4c05c

> Sometimes the return queryset has duplicate elements (is this true?)

Yes, it's possible.

> What kind of situations would cause `finished` to equal `False` here?

An incomplete iteration would cause `finished` to equal `False`. For example, if the calling code did `next(iter(queryset))` then `items1` would only contain a single item even though `queryset` actually contains more than one item. In that case,  since we cannot depend on ordering, the best we can do is verify that all items in `ids1` (mptt results) are also in `ids2` (cte results).

Reply:
gotcha, thanks for the explanation!